### DISCUSSION TITLE
Route file is broken when use a method to define return

### DISCUSSION THREAD

**Discussion by Mendelski** [laravel/framework#51169] [Q&A]
### Laravel Version

10.45.1

### PHP Version

8.2

### Database Driver & Version

PgSQL

### Description

i tryied to make a return on a route only for debug, using a function on api.php just like 
```
Route::middleware('auth:sanctum')->get('/user', function (Request $request) {
    return $request->user();
});
```
is a little bigger, but only a query and a return. 

and i have this error. 

```
Call to undefined method Closure::__set_state()

  at bootstrap\cache\routes-v7.php:2021
    2017▕         array (
    2018▕           0 => 'api',
    2019▕         ),
    2020▕         0 => 
  ➜ 2021▕         \Closure::__set_state(array(
    2022▕         )),
    2023▕         'uses' => 'O:55:"Laravel\\SerializableClosure\\UnsignedSerializableClosure":1:{s:12:"serializable";O:46:"Laravel\\SerializableClosure\\Serializers\\Native":5:{s:3:"use";a:0:{}s:8:"function";s:205:"function () {
    2024▕         $fields = [\'idEnvio\', \'type\', \'amount\', \'status\'];
    2025▕

```
even if i try delete my new route, the error keep. 

**My solution is delete cache/route-v7 file and rebuild caches, and now works fine. (using a function inside controller)**
 
 

### Steps To Reproduce

a route like this 

Route::get('/user', function (Request $request) {
     $t = Transaction::query()->select($fields)->latest()->paginate(10);
        return response()->json($t);
});

**Comment by joy2362** [comment]
Hello @Mendelski  issue is in your select you add **$fields** but that variable not init in that code
